apps/finance/app/src/lib/conversion-utils.js

```diff
@@ -1,66 +1,34 @@
 import { useEffect, useRef, useState } from 'react'
 import BN from 'bn.js'
+import { formatTokenAmount } from '@aragon/ui'
 
 const CONVERT_API_RETRY_DELAY = 2 * 1000
 const CONVERT_API_RETRY_DELAY_MAX = 60 * 1000
-
-const USD_DECIMALS = new BN('2')
+const CONVERT_PRECISION = 9
 
 function convertRatesUrl(symbolsQuery) {
   return `https://min-api.cryptocompare.com/data/price?fsym=USD&tsyms=${symbolsQuery}`
 }
 
-function formatConvertRate(convertRate, decimals) {
-  const [whole = '', dec = ''] = convertRate.split('.')
-  const parsedWhole = whole.replace(/^0*/, '')
-  const parsedDec = dec.replace(/0*$/, '')
-  // parsedWhole could be empty,
-  // so in this case, we wanna remove leading zeros.
-  const fullyParsedDec = parsedWhole ? parsedDec : parsedDec.replace(/^0*/, '')
-
-  // Even if we remove leading zeroes from the decimal
-  // part, we want to count as if we "shifted" them
-  const decimalsToShift = decimals.sub(new BN(parsedDec.length.toString()))
-  // Apart from always considering the USD decimals (2),
-  // if there's the strange case that the above is negative,
-  // we take it as a carry as we know we already shifted to far,
-  // and will compensate by shifting the token amount by this much
-  const carryAmount =
-    decimalsToShift.toNumber() < 0
-      ? decimalsToShift.add(USD_DECIMALS)
-      : USD_DECIMALS
-  // The remaining total amount to shift through bn.js to avoid overflow.
-  const amountToShift = new BN('10').pow(
-    decimalsToShift.toNumber() > 0 ? decimalsToShift : new BN('0')
-  )
-
-  // Finish shifting the whole number through BN.js to avoid overflow,
-  return [
-    new BN(`${parsedWhole}${fullyParsedDec}`).mul(amountToShift),
-    carryAmount,
-  ]
-}
-
-export function getConvertedAmount(amount, convertRate, decimals) {
-  const [formattedConvertRate, carryAmount] = formatConvertRate(
-    convertRate.toString(),
-    decimals
+/**
+ * Convert an amount into another one using a rate.
+ *
+ * @param {BigInt|string|number} amount amount to convert
+ * @param {BigInt|string|number} decimals number of decimals for the amount
+ * @param {string|number} rate the rate to use for the conversion
+ * @param {Object} [options] options passed to formatTokenAmount()
+ * @returns {string} the formatted amount converted
+ */
+export function convertAmount(amount, decimals, rate, options) {
+  amount = new BN(String(amount))
+  decimals = parseInt(String(decimals), 10)
+  return formatTokenAmount(
+    amount
+      .mul(new BN(10).pow(new BN(CONVERT_PRECISION)))
+      .mul(new BN(rate * 10 ** CONVERT_PRECISION)),
```

Comment:
Hmm, not entirely sure on this; you're going to get an error if `rate * 10 ** CONVERT_PRECISION` becomes an imprecise number.